### PR TITLE
Added loading indicator and set default imageUrl to placeholder gif

### LINKED ISSUE #41: Add a loading indicator when waiting for API response
As a user, I want to see some indication of loading while waiting for the image to be generated from the API response

AC: 
 - loading indicator appears after submitting a text-to-image search
 - loading indicator disappears when response is received

### PR DESCRIPTION
**Pull request by srimel**
Let me know if the URL change I made is blocking anyone and I can roll that back. I made it such that the API is only called when the user submits a search. On page reloading, the default is still the giphy placeholder from before. 

For the loading bar, I'm using a gif which is loaded into the Player component and useState to conditionally render the Player component with the loading gif.

### CODE REVIEW

File: artscii/src/components/main/Main.js [hunk 1]
```diff
@@ -43,7 +46,11 @@ function Main() {
       <div className='main'>
           <h2>Searching for:</h2>
           <h2>{displayText}</h2>
-          <Player url={url} search={searchParam} playerMode={playerMode}/>
+          {loading ? (
+            <Player url='https://media.giphy.com/media/KKCuBooszlPG0/giphy.gif' playerMode={playerMode} />
```
Comment (unfinishedideas) [inline artscii/src/components/main/Main.js]: nit: do we want to rely on this hosted gif or import one from the respository?

**Review by unfinishedideas — APPROVED** [review]
Looks good! Just wondering if we'd rather import a gif from the project in case that loading gif URL goes down

**Comment by anamika8** [comment]
Looks cool to me!

**Review by anamika8 — APPROVED** [review]

**Review by elizasohn — APPROVED** [review]
looks good! I think in the future maybe setting the css for the width and height smaller so it looks less pixelated or creating our own loading bar would be nice!

File: artscii/src/components/main/Main.js [hunk 2]
```diff
@@ -32,9 +34,11 @@ function Main() {
             .then(imageUrl => {
                 console.log(`Image URL received in UI - ${imageUrl}`)
                 setImageUrl(imageUrl)
+                setLoading(false);
             })
             .catch(err => {
                 console.log("error encountered = " + err);
+                setLoading(false);
             });
```
Comment (contolini) [inline artscii/src/components/main/Main.js]: FWIW you can avoid this duplication with a [`finally`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/finally):

```suggestion
            })
            .catch(err => {
                console.log("error encountered = " + err);
            })
            .finally(() => {
                setLoading(false);
            });
```

ok love you bye have a nice day

Comment (unfinishedideas) [reply artscii/src/components/main/Main.js]: https://youtu.be/hvHPv1go75o

Comment (srimel) [reply artscii/src/components/main/Main.js]: @contolini  thanks for the suggestion! I updated the PR

**Pull request merged**